pkg/jobmanager/job.go

```diff
@@ -50,11 +55,21 @@ func newJob(owner, cmd string, args []string) *Job {
 		waitChan: make(chan struct{}),
 	}
 
-	// Set the process to run in it's own pgid.
-	// NOTE: Will probably update this to run in it's own session once we
-	// get to the cgroups/namespace implementation.
 	j.cmd.SysProcAttr = &syscall.SysProcAttr{
+		// Set the process to run in it's own pgid.
+		// NOTE: Will probably update this to run in it's own session once we
+		// get to the cgroups/namespace implementation.
 		Setpgid: true,
+
+		// Create the job in namespaces for isolation.
+		Cloneflags: syscall.CLONE_NEWPID | // PID namespace.
+			syscall.CLONE_NEWNS | // Mount namespace.
+			syscall.CLONE_NEWNET, // Network namespace.
+
+		// Make use of clone3 cgroup arg.
+		UseCgroupFD: true,
+
+		PidFD: new(int),
```

Comment:
This doesn't seem to be referenced anywhere else - crucially, it doesn't get closed either.

Reply:
Indeed, not needed, removed.